crates/composer_api/src/lib.rs

```diff
@@ -12,8 +12,57 @@ use std::{
 
 pub const DEFAULT_SERVER_ADDRESS: &str = "localhost:8888";
 
+/// Composer expects `EventMessage` as the incoming probe data.
 #[derive(Serialize, Deserialize, Debug)]
-pub enum Event {
+pub struct EventMessage {
+    /// List of events a probe collected during a specific time window. For a probe generating
+    /// high-frequency events e.g. more than a hundred per second, it's recommended to buffer and
+    /// pack multiple events into a `EventMessage` to avoid overflowing the socket and reduce
+    /// packet overhead.
+    pub events: Vec<Event>,
+}
+
+impl EventMessage {
+    pub fn new(events: Vec<Event>) -> Self {
+        Self { events }
+    }
+
+    pub fn with_event(event: Event) -> Self {
```

Comment:
Naming nit: `from_event()`? (`with_event()` sounds to me as if event is something optional in a builder pattern, but this is something slightly different).